nationalarchives/ds-wagtail
# Generate URLs for What's On "remove filter" buttons

No ticket

## About these changes

Branched from https://github.com/nationalarchives/ds-wagtail/pull/1409, which includes the markup for the buttons.

This PR adds logic to generate URLs for the active filter buttons, such that when the links (buttons) are clicked, that filter is removed.

## How to check these changes

1. Create What's On Listing Page
2. Create event pages as children
3. View the listing page
4. With no filters applied, there should be no "active filter" buttons, including the "clear filters" button
5. Apply filters, test that the "clear filters" button appears and when clicked, removes all filters from the listing
6. Apply filters, test that the generated buttons remove the corresponding filter from the listing


## Before assigning to reviewer, please make sure you have


## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)
